src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeFormat.cs

```diff
@@ -1112,6 +1112,233 @@ private static StringBuilder FormatStringBuilder(DateTime dateTime, ReadOnlySpan
             return FormatCustomized(dateTime, format, dtfi, offset, result: null);
         }
 
+        internal static bool IsValidCustomDateFormat(ReadOnlySpan<char> format, bool allowThrow)
+        {
+            int length = format.Length;
+            int i = 0;
+
+            while (i < length)
+            {
+                switch (format[i])
+                {
+                    case '\\':
+                        if (i == length - 1)
+                        {
+                            if (allowThrow)
+                            {
+                                throw new FormatException(SR.Format_InvalidString);
+                            }
+
+                            return false;
+                        }
+                        i += 2;
+                        break;
+
+                    case '\'':
+                    case '"':
+                        char quoteChar = format[i++];
+                        while (i < length && format[i] != quoteChar)
+                        {
+                            i++;
+                        }
```

Comment:
Maybe use IndexOf() to allow optimizations?

Below too.

Reply:
I don't think so calling other method would be faster in this case.